packages/core/src/tracing/idleSpan.ts

```diff
@@ -108,6 +109,20 @@ export function startIdleSpan(startSpanOptions: StartSpanOptions, options: Parti
   const previousActiveSpan = getActiveSpan();
   const span = _startIdleSpan(startSpanOptions);
 
+  function _endSpan(timestamp: number = timestampInSeconds()): void {
+    // Ensure we end with the last span timestamp, if possible
+    const spans = getSpanDescendants(span).filter(child => child !== span);
+
+    const latestEndTimestamp = spans.length ? Math.max(...spans.map(span => spanToJSON(span).timestamp || 0)) : 0;
+
+    // If the timestamp is smaller than the latest end timestamp of a span, we still want to use it
+    const endTimestamp = latestEndTimestamp
+      ? Math.min(spanTimeInputToSeconds(timestamp), latestEndTimestamp)
+      : timestamp;
```

Comment:
Something looks off here. In one case we take `spanTimeInputToSeconds(timestamp)`, in the other case we take `timestamp`.

Reply:
this is fine here because we only need to convert it to a seconds timestamp for `Math.min()` to work reliably, otherwise we can just pass the input through whatever it is!

Reply:
Ok, I assume because we convert it to the right unit downstream somewhere? Would you mind adding a (short) comment? This might confuse the next person coming across this ^^

Reply:
updated the whole code block a bit to hopefully clarify thigns!